### ISSUE TITLE
Improving the Logstash Logging subsystem (aka making it less versbose)

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#4621]
A common pain point of logstash monitoring is processing its logs, they could become very big and verbose. I propose a set of actions that will help improve the overall logging system.

* Revisit logging with in logstash and their plugins to make it as accurate as possible, removing unnecessary log.
* Enable trace logs, they should be turn on by default and only available when debugging. This will help getting accurate logs only for parts under investigation.
* Let users decide to get only logs generated by a selected groups of components.

The intention of this issue is to discuss and generate ideas to have a clever logging infra, there related issues focused on implementation details.

Relates to #4548

**Comment by suyograo** [comment]
+1 on reviewing existing logs for log level. A lot of the existing log messages can be trace instead of debug. This work should happen in conjunction with #4548

**Comment by jordansissel** [comment]
+1 @suyograo's comment. As a short-term solution, we can add a trace level to Cabin pretty easily.

**Comment by purbon** [comment]
I agree with you two, we've several examples of tracing log, that will let us focus on a more clean logging info. 

@suyograo one is for sure related with the other, one thing is to know what to clean and revisiting our logging strategies and another is to have tools that let us implement it, boths should happen related.

**Comment by jsvd** [comment]
there's no doubt we'd benefit immensely from hierarchical logging configuration, specially now with the logstash.yml file, we could have a logging section:

```yaml
logging:
  pipeline: info
  event: warn
  agent: debug
  plugin:
     input: info
     filter: debug
     output: trace
```

that aside....+1 on all suggestions

**Comment by andrewvc** [comment]
Regarding the hierarchy stuff, SLF4J already supports this IIRC. Can't you
pick which 'package' the logs come from? I don't know how this works with
ruby classes.

**Comment by purbon** [comment]
Agree with that, been reading today docs about slf4j and have a ton of nice
features. I think i will start and experiment to see direct points of
improvement.

/purbon
